### PR TITLE
e2e Tutorial

### PR DESCRIPTION
**Pull request by ncoop57**
This PR contains a e2e tutorial for how create a drawing app and deploy it to railway

### CODE REVIEW

**Comment by ncoop57** [comment]
Need to figure out the issue with railway deployment b4 merging. Still investigating

File: nbs/tutorials/01_quickdraw.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+rooms = db.t.rooms
+if rooms not in db.t:
+    rooms.create(id=int, name=str, created_at=str, pk='id')
+Room = rooms.dataclass()
+
+@patch
+def __xt__(self:Room):
+    return Li(A(self.name, href=f"/rooms/{self.id}"))
+
+app = FastHTML(hdrs=(picolink,))
```
Comment (jph00) [inline nbs/tutorials/01_quickdraw.py:14]: You can replace lines 4-15 with a single call to fast_app, including using `render` (although you'll still need to define that function of course). E.g
```python
app,rt,rooms.Room = FastHTML('data/drawapp.db', render=render, id=int, name=str, created_at=str, pk='id')
```

**Review by jph00 — COMMENTED** [review]
Looking great!

File: nbs/tutorials/01_quickdraw.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    room.created_at = datetime.now().isoformat()
+    new_room = rooms.insert(room)
+    return Li(A(new_room.name, href=f"/rooms/{new_room.id}"))
+
+@rt("/rooms/{id}")
+async def get(id:int):
+    room = rooms[id]
+    return Titled(f"Room: {room.name}", H1(f"Welcome to {room.name}"), A(Button("Leave Room"), href="/"))
+
+if __name__ == "__main__": run_uv()
```
Comment (jph00) [inline nbs/tutorials/01_quickdraw.py:36]:
```suggestion
run_uv()
```

You can change this in all of them.

File: nbs/tutorials/01_quickdraw.py [hunk 3]
```diff
@@ -0,0 +1,28 @@
+from fasthtml.common import *
+from datetime import datetime
+
+def render(room):
+    return Li(A(room.name, href=f"/rooms/{room.id}"))
+
+app,rt,rooms,Room = fast_app('data/drawapp.db', hdrs=(picolink,), render=render, id=int, name=str, created_at=str, pk='id')
```
Comment (jph00) [inline nbs/tutorials/01_quickdraw.py:7]: picolink is included by default.
```suggestion
app,rt,rooms,Room = fast_app('data/drawapp.db', render=render, id=int, name=str, created_at=str, pk='id')
```

File: nbs/tutorials/01_quickdraw.py [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+@rt("/")
+def get():
+    create_room = Form(Input(id="room-name", name="name", placeholder="New Room Name"),
```
Comment (jph00) [inline nbs/tutorials/01_quickdraw.py:11]: name defaults to same as id, so if you pick a good id often you can skip name

File: nbs/tutorials/01_quickdraw.py [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+                       Button("Create Room"),
+                       hx_post="/rooms", hx_target="#rooms-list", hx_swap="afterbegin")
+    rooms_list = Ul(*rooms(order_by='id DESC'), id='rooms-list')
+    return Titled("QuickDraw", create_room, rooms_list)
+
+@rt("/rooms")
+async def post(room:Room):
+    room.created_at = datetime.now().isoformat()
+    new_room = rooms.insert(room)
```
Comment (jph00) [inline nbs/tutorials/01_quickdraw.py:20]:
```suggestion
    return rooms.insert(room)
```

The render function should handle this.

**Comment by jph00** [comment]
@ncoop57 please let me know when it's ready for re-review.

**Comment by ncoop57** [comment]
@jph00 okay should be all good to go for re-review! Thanks!

**Comment by jph00** [comment]
Nice! :D

**Pull request merged**